### PR TITLE
[AWS] Always run the Data_Sync_Complete job after the data sync

### PR DESCRIPTION
**Pull request by issyl0**
- This job wasn't running, causing alerts.

### CODE REVIEW

File: hieradata_aws/class/jenkins.yaml [hunk 1]
```diff
@@ -53,6 +53,7 @@ govuk_jenkins::config::user_permissions:
 govuk_jenkins::config::banner_string: 'AWS Deploy'
 
 govuk_jenkins::job_builder::jobs:
+  - govuk_jenkins::job::data_sync_complete
```
Comment (surminus) [inline hieradata_aws/class/jenkins.yaml]: I think this will need to be `govuk_jenkins::job::data_sync_complete_integration`

Comment (issyl0) [reply hieradata_aws/class/jenkins.yaml]: Ah!

**Comment by issyl0** [comment]
@surminus Your PR https://github.com/alphagov/govuk-puppet/pull/6793 adds `run_rake_task` to the list of jobs, so that should be merged first?

**Comment by surminus** [comment]
Merged!

File: modules/govuk_jenkins/templates/jobs/data_sync_complete_integration.yaml.erb [hunk 2]
```diff
@@ -12,9 +12,15 @@
     builders:
        - shell: |
            # Clear the Sidekiq queues to stop any residual 'stale' jobs running
+           <%- if scope.lookupvar('::aws_migration') %>
+           for host in $(govuk_node_list -c backend); do
```
Comment (surminus) [inline modules/govuk_jenkins/templates/jobs/data_sync_complete_integration.yaml.erb]: I think we only need to run this once, so I would add `--single-node` to the `govuk_node_list` command.

**Comment by surminus** [comment]
One small comment, but otherwise looks OK!

File: modules/govuk_jenkins/templates/jobs/data_sync_complete_integration.yaml.erb (hunk 2 above)
Comment (issyl0) [reply modules/govuk_jenkins/templates/jobs/data_sync_complete_integration.yaml.erb]: In Carrenza on the redis hosts it did `for host in ...`, and there were multiple hosts... I assumed we'd need to run it on every host, but now you mention it if it's connecting to the central Redis then it's fine.

(I discovered yesterday that `redis-cli` isn't on backend machines in AWS, which makes running it harder (!), and I can't find anywhere that installs it in Puppet, so we'll have to fix that.)

Comment (surminus) [reply modules/govuk_jenkins/templates/jobs/data_sync_complete_integration.yaml.erb]: Yup, we only have essentially single Redis endpoint, whereas in Carrenza we (confusingly) have two separate instances.

**Pull request merged**